subset.py

```diff
@@ -178,11 +181,26 @@ def __subset_upstream(self, target, processed_tables, relationships):
             # filter it down in the target database
             table_columns = self.__db_helper.get_table_columns(
                 table_name(target), schema_name(target), self.__source_conn)
-            clauses = ['{} IN (SELECT {} FROM {})'.format(
-                columns_tupled(kc['fk_columns']),
-                columns_joined(kc['target_columns']),
-                fully_qualified_table(mysql_db_name_hack(kc['target_table'],
-                                                         self.__destination_conn))) for kc in relevant_key_constraints]
+
+            clauses = []
+            # This is where we look at the FK augmentation.
+            for kc in relevant_key_constraints:
+                if kc.get('polymorphic_type'):
+                    clause = '{} IN (SELECT {} FROM {}) AND {}'.format(
+                        columns_tupled(kc['fk_columns']),
+                        columns_joined(kc['target_columns']),
+                        fully_qualified_table(mysql_db_name_hack(kc['target_table'], self.__destination_conn)),
+                        kc['polymorphic_type']
+                    )
+                    clauses.append(clause)
+                else:
+                    clause = '{} IN (SELECT {} FROM {})'.format(
+                        columns_tupled(kc['fk_columns']),
+                        columns_joined(kc['target_columns']),
+                        fully_qualified_table(mysql_db_name_hack(kc['target_table'], self.__destination_conn))
+                    )
+                    clauses.append(clause)
```

Comment:
Evaluate whether or not a foreign key constraints have a polymorphic relationship define and create the appropriate query.